tests/trainer/logging_/test_train_loop_logging.py

```diff
@@ -712,3 +712,35 @@ def training_step(self, *args):
     model = TestModel()
     with pytest.raises(MisconfigurationException, match=r'reduce_fx={min,max,mean,sum}\)` are currently supported'):
         trainer.fit(model)
+
+
+def test_sanity_metrics_are_reset_and_dont_appear_in_prog_bar(tmpdir):
+
+    class TestModel(BoringModel):
+
+        def validation_step(self, batch, batch_idx):
+            output = super().validation_step(batch, batch_idx)
+            if self.trainer.sanity_checking:
+                self.log("val_loss", output["x"], prog_bar=True, logger=True)
+            return output
+
+        def training_step(self, batch, batch_idx):
+            loss = super().training_step(batch, batch_idx)
+            if batch_idx == 0:
+                pass
```

Comment:
```suggestion
            if batch_idx == 0:
```
probably better to remove it 😄

Reply:
Done !